test/test_fcntl.rs

```diff
@@ -180,3 +180,36 @@ mod linux_android {
         assert_eq!(100, read(fd, &mut buf).unwrap());
     }
 }
+
+#[cfg(any(target_os = "linux",
+          target_os = "android",
+          target_os = "emscripten",
+          target_os = "fuchsia",
+          any(target_os = "wasi", target_env = "wasi"),
+          target_env = "uclibc",
+          target_env = "freebsd"))]
+mod test_posix_fadvise {
+
+    use tempfile::NamedTempFile;
+    use std::os::unix::io::{RawFd, AsRawFd};
+    use nix::errno::Errno;
+    use nix::fcntl::*;
+    use nix::unistd::pipe;
+
+    #[test]
+    fn test_posix_fadvise_success() {
```

Comment:
Thanks to the module, "posix_fadvise" is redundant in the name of these test functions.